packages/zosfiles/src/methods/copy/Copy.ts

```diff
@@ -93,6 +103,75 @@ export class Copy {
         }
     }
 
+    /**
+     * Private function that checks if a dataset is type PDS
+    **/
+    public static async isPDS(
+        session: AbstractSession,
+        dataSetName: string
+    ): Promise<boolean> {
+        try {
+            const response = await List.dataSet(session, dataSetName, {attributes: true});
+            const dsorg = response.apiResponse.items[0].dsorg;
+            return dsorg.startsWith("PO");
+        }
+        catch(error) {
+            Logger.getAppLogger().error(error);
+            throw error;
+        }
+    }
+
+    /**
+     * Copy the members of a Partitioned dataset into another Partitioned dataset
+     *
+     * @param {AbstractSession}   session        - z/OSMF connection info
+     * @param {IDataSet}          toDataSet      - The data set to copy to
+     * @param {IDataSetOptions}   options        - Options
+     *
+     * @returns {Promise<IZosFilesResponse>} A response indicating the status of the copying
+     *
+     * @throws {ImperativeError} Data set name must be specified as a non-empty string
+     * @throws {Error} When the {@link ZosmfRestClient} throws an error
+     *
+     * @see https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.1.0/com.ibm.zos.v2r1.izua700/IZUHPINFO_API_PutDataSetMemberUtilities.htm
+     */
+
+    public static async copyPDS (
+        session: AbstractSession,
+        fromPds: string,
+        toPds: string
+    ): Promise<IZosFilesResponse> {
+        try {
+            const sourceResponse = await List.allMembers(session, fromPds);
+            const sourceMemberList: Array<{ member: string }> = sourceResponse.apiResponse.items;
+
+            if(sourceMemberList.length == 0) {
+                return {
+                    success: true,
+                    commandResponse: `Source dataset (${fromPds}) - ` + ZosFilesMessages.noMembersFound.message
+                };
+            }
+
+            const downloadDir = path.join(tmpdir(), fromPds);
+            await Download.allMembers(session, fromPds, {directory:downloadDir});
+            const uploadFileList: string[] = ZosFilesUtils.getFileListFromPath(downloadDir);
+
+            for (const file of uploadFileList) {
+                const uploadingDsn = `${toPds}(${ZosFilesUtils.generateMemberName(file)})`;
+                const uploadStream = IO.createReadStream(file);
+                await Upload.streamToDataSet(session, uploadStream, uploadingDsn);
+            }
+            fs.rmSync(downloadDir, {recursive: true});
+            return {
+                success:true,
+                commandResponse: ZosFilesMessages.datasetCopiedSuccessfully.message
+            };
+        }
+        catch (error) {
```

Comment:
Here is an example of how a `truncation error`, i.e. the source `lrecl` is greater than the target `lrecl`, might look like
![image](https://github.com/user-attachments/assets/b0787dcb-012f-4bd1-9848-3ab9d21622aa)

---

And it will still upload the member, just that it will truncate the data. Then stops uploading members.
![image](https://github.com/user-attachments/assets/c43c5560-c8ed-4068-8af8-ef4fe2efde50)

---

Curious if we could get some progress bars (ideally with the member name that's being copied) 😋 
That will give people a sense of how much might be left, or if a bad error occurs, they know where the copy operation stopped 😋

Reply:
Might it be possible to add the name of the member to error during the catch block before we rethrow it? Maybe there are problems with this idea. Just asking that you evaluate the idea.